src/outputs/parthenon_hdf5.cpp

```diff
@@ -274,45 +279,129 @@ void PHDF5Output::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) {
-        writeH5AF64("Time", &(tm->time), localDSpace, myDSet);
+        writeH5AI32("NCycle", &(tm->ncycle), localDSpace, infoDSet);
+        writeH5AF64("Time", &(tm->time), localDSpace, infoDSet);
       }
-      writeH5AI32("NumDims", &pm->ndim, localDSpace, myDSet);
-      writeH5AI32("NumMeshBlocks", &pm->nbtotal, localDSpace, myDSet);
-      writeH5AI32("MaxLevel", &max_level, localDSpace, myDSet);
+      writeH5AI32("NumDims", &pm->ndim, localDSpace, infoDSet);
+      writeH5AI32("NumMeshBlocks", &pm->nbtotal, localDSpace, infoDSet);
+      writeH5AI32("MaxLevel", &max_level, localDSpace, infoDSet);
       // write whether we include ghost cells or not
       int iTmp = (output_params.include_ghost_zones ? 1 : 0);
-      writeH5AI32("IncludesGhost", &iTmp, localDSpace, myDSet);
+      writeH5AI32("IncludesGhost", &iTmp, localDSpace, infoDSet);
       // write number of ghost cells in simulation
       iTmp = Globals::nghost;
-      writeH5AI32("NGhost", &iTmp, localDSpace, myDSet);
+      writeH5AI32("NGhost", &iTmp, localDSpace, infoDSet);
       writeH5ASTRING("Coordinates", std::string(first_block.coords.Name()), localDSpace,
-                     myDSet);
+                     infoDSet);
 
       hsize_t nPE = Globals::nranks;
       writeH5AI32("BlocksPerPE", nblist.data(),
-                  H5S::FromHIDCheck(H5Screate_simple(1, &nPE, NULL)), myDSet);
+                  H5S::FromHIDCheck(H5Screate_simple(1, &nPE, NULL)), infoDSet);
 
       // open vector space
       // write mesh block size
       int meshblock_size[3] = {nx1, nx2, nx3};
       const hsize_t xDims[1] = {3};
       writeH5AI32("MeshBlockSize", meshblock_size,
-                  H5S::FromHIDCheck(H5Screate_simple(1, xDims, NULL)), myDSet);
-    }
+                  H5S::FromHIDCheck(H5Screate_simple(1, xDims, NULL)), infoDSet);
+
+      // RootGridDomain - float[9] array with xyz mins, maxs, rats (dx(i)/dx(i-1))
+      Real rootGridDomain[9] = {
+          pm->mesh_size.x1min, pm->mesh_size.x1max, pm->mesh_size.x1rat,
+          pm->mesh_size.x2min, pm->mesh_size.x2max, pm->mesh_size.x2rat,
+          pm->mesh_size.x3min, pm->mesh_size.x3max, pm->mesh_size.x3rat};
+      const hsize_t rootGridDomain_size[1] = {9};
+      writeH5AF64("RootGridDomain", rootGridDomain,
+                  H5S::FromHIDCheck(H5Screate_simple(1, rootGridDomain_size, NULL)),
+                  infoDSet);
+
+      // RootGridSize - int[3] number of cells on the root grid
+      int rootGridSize[3] = {pm->mesh_size.nx1, pm->mesh_size.nx2, pm->mesh_size.nx3};
+      const hsize_t rootGridSize_size[1] = {3};
+      writeH5AI32("RootGridSize", rootGridSize,
+                  H5S::FromHIDCheck(H5Screate_simple(1, rootGridSize_size, NULL)),
+                  infoDSet);
+
+      // BoundaryConditions
+      std::vector<std::string> boundaryConditions(BOUNDARY_NFACES);
+      for (int i = 0; i < BOUNDARY_NFACES; i++) {
+        boundaryConditions[i] = GetBoundaryString(pm->mesh_bcs[i]);
+      }
```

Comment:
In general I believe it will be more performant to call reserve with push_back because you get to avoid a default construction call. Not that it will really matter here.